src/components/CreateTargetForm/index.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { func, shape, number, string, oneOfType, bool } from 'prop-types';
+import { View, Text, TouchableOpacity, Platform } from 'react-native';
+import { LOADING } from '@rootstrap/redux-tools';
+import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
+
+import ErrorView from 'components/common/ErrorView';
+import Button from 'components/common/Button';
+import Input from 'components/common/Input';
+import strings from 'locale';
+import useForm from 'hooks/useForm';
+import useValidation from 'hooks/useValidation';
+import createTargetValidations from 'validations/createTargetValidations';
+import useCreateTargetForm from 'hooks/useCreateTargetForm';
+import useTextInputProps from 'hooks/useTextInputProps';
+import { BLACK } from 'constants/colors';
+import common from 'constants/commonStyles';
+import fonts from 'constants/fonts';
+import styles from './styles';
+
+const FIELDS = {
+  areaLength: 'areaLength',
+  title: 'title',
+};
+const { TARGET } = strings;
+
+const CreateTargetForm = ({ onSubmit, onShowTopics, onHide, topic }) => {
+  const validator = useValidation(createTargetValidations);
+
+  const { values, errors, handleValueChange, handleSubmit, handleBlur } = useForm(
+    {
+      onSubmit,
+      validator,
+      validateOnBlur: true,
+    },
+    [onSubmit],
+  );
+
+  const {
+    handleOnPress,
+    targetError,
+    targetStatus,
+    getTopicsError,
+    getTopicsStatus,
+    topicError,
+  } = useCreateTargetForm(topic, TARGET, handleSubmit);
+
+  const inputProps = useTextInputProps(handleValueChange, handleBlur, values);
+
+  const topicText = () => {
+    if (getTopicsError) return TARGET.topicsError;
+    if (getTopicsStatus === LOADING) return TARGET.loadingTopics;
+    return (topic && topic.label) || TARGET.topicPlaceholder;
+  };
+
+  return (
+    <>
+      <TouchableOpacity style={styles.mainContainer} onPress={onHide} />
+      <KeyboardAwareScrollView keyboardShouldPersistTaps="handled" style={styles.container}>
+        <View style={styles.inputContainer}>
+          <Input
+            label={TARGET.area}
+            testID="area-input"
+            keyboardType="numeric"
+            error={errors[FIELDS.areaLength]}
+            invalid={targetError || errors[FIELDS.areaLength]}
+            short={false}
+            placeholder={TARGET.areaPlaceholder}
+            placeholderTextColor={BLACK}
```

Comment:
Why put placeholder text black? I think it is confusing because you may think you already fill the field when you haven't. Isn't the default grey color better?

Reply:
I agree, but I just did it to make it look like the designs